File: 1 Algorithmic Toolbox/Homeworks/3 Greedy Algorithms/1 Money Change/C++/change.cpp
#include <iostream>

int get_change(int m) {
	int typesOfCoins[3] = {10, 5, 1};
	int counterOfCoins(0), curType(0);

	while (m != 0) {
		if (m / typesOfCoins[curType] != 0) {
			counterOfCoins++;
			m -= typesOfCoins[curType];
		}
		else curType++;
	}

	return counterOfCoins;
}

int main() {
	int m;
	std::cin >> m;
	std::cout << get_change(m) << '\n';
}

File: 1 Algorithmic Toolbox/Homeworks/3 Greedy Algorithms/1 Money Change/change.py
# Uses python3


def get_change(m):
    types_of_coins = [1, 5, 10]
    counter_of_coins = 0
    while m != 0:
        if m // max(types_of_coins) != 0:
            counter_of_coins += 1
            m -= max(types_of_coins)
        else:
            types_of_coins.remove(max(types_of_coins))
    return counter_of_coins


if __name__ == '__main__':
    m = int(input())
    print(get_change(m))
